ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -1259,9 +1259,15 @@ void processReplicaMetadataResponse(Set<MessageInfo> missingRemoteStoreMessages,
     for (MessageInfo messageInfo : messageInfoList) {
       BlobId blobId = (BlobId) messageInfo.getStoreKey();
       if (remoteReplicaInfo.getLocalReplicaId().getPartitionId().compareTo(blobId.getPartition()) != 0) {
-        throw new IllegalStateException(
-            "Blob id " + blobId.getID() + " is not in the expected partition Actual partition " + blobId.getPartition()
-                + " Expected partition " + remoteReplicaInfo.getLocalReplicaId().getPartitionId());
+        String errorMessage =
```

Comment:
Won't this send get request for blob ids ?
And when we get and put these blob in store, store would throw exception as partition id will not match.